src/mlpack/core/optimizers/adam/nadam_update.hpp

```diff
@@ -0,0 +1,169 @@
+/**
+ * @file nadam_update.hpp
+ * @author Sourabh Varshney
+ *
+ * Nadam update rule. Nadam is an optimizer that combines the effect of Adam and 
+ * NAG to the gradient descent to improve its Performance.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_ADAM_NADAM_UPDATE_HPP
+#define MLPACK_CORE_OPTIMIZERS_ADAM_NADAM_UPDATE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * Nadam is an optimizer that combines the Adam and NAG.
+ *
+ * For more information, see the following.
+ *
+ * @code
+ * @techreport{Dozat2015,
+ *   title       = {Incorporating Nesterov momentum into Adam},
+ *   author      = {Timothy Dozat},
+ *   institution = {Stanford University},
+ *   address     = {Stanford},
+ *   year        = {2015}
+ * }
+ * @endcode
+ */
+class NadamUpdate
+{
+ public:
+  /**
+   * Construct the Nadam update policy with the given parameters.
+   *
+   * @param epsilon The epsilon value used to initialise the squared gradient
+   *        parameter.
+   * @param beta1 The smoothing parameter.
+   * @param beta2 The second moment coefficient
```

Comment:
Can you add a comment for the ``scheduleDecay`` parameter? Also another super pedantic style issue; can you tab the initialization list twice so that it looks like:

```c++
NadamUpdate(const double epsilon = 1e-8,
            const double beta1 = 0.9,
            const double beta2 = 0.99,
            const double scheduleDecay = 4e-3) :
    epsilon(epsilon),
    beta1(beta1),
    beta2(beta2),
    scheduleDecay(scheduleDecay),
    iteration(0),
    cumBeta1(1)
```

Reply:
Done that